### PR TITLE
Abandon ValueTask

### PR DESCRIPTION
**Pull request by bruno-garcia**
The caching code hangs, has flaky and and fails on Mono completely on Unity and Xamarin.

Attempting to figure out the culprit. First try: Get rid of ValueTask which makes it fucking impossible to debug, even on `net5.0` target. At least on Rider on a macOS.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,7 +19,7 @@
 ## 3.0.0-alpha.4
 
 * Add the client user ip if both SendDefaultPii and IsEnvironmentUser are set. (#1015) @lucas-zimerman
-* Replace Task with ValueTask where possible. (#564) @Tyrrrz
+* Replace Task with Task where possible. (#564) @Tyrrrz
```
Comment (bruno-garcia) [inline CHANGELOG.md:22]:
```suggestion
* Replace Task with ValueTask where possible. (#564) @Tyrrrz
```

File: src/Sentry/Internal/Http/CachingTransport.cs [hunk 2]
```diff
@@ -65,34 +65,38 @@ public CachingTransport(ITransport innerTransport, SentryOptions options)
                 }
             }
 
-            _worker = Task.Run(async () =>
+            _worker = Task.Run(CachedTransportBackgroundTask);
+        }
+
+        private async Task? CachedTransportBackgroundTask()
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Http/CachingTransport.cs:71]: Method is easier to visualize in the stacktrace while debugging on rider.
Shouldn't return nullable but

**Comment by bruno-garcia** [comment]
@Tyrrrz this PR was tested by @lucas-zimerman on Xamarin and seem to fix the hanging issues.
I'll go ahead with this while we continue to debug this stuff. We can discuss later if you think we should rethink this, like use ValueTask on some things but not others.

I don't really know what was the culprit to the flakyness, might be a runtime bug

File: test/Sentry.Tests/Extensibility/DisabledHubTests.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public class DisabledHubTests
 
         [Fact]
         public void ConfigureScopeAsync_ReturnsCompletedTask()
-            => Assert.Equal(default, DisabledHub.Instance.ConfigureScopeAsync(null));
+            => Assert.Equal(Task.CompletedTask, DisabledHub.Instance.ConfigureScopeAsync(null!));
```
Comment (lucas-zimerman) [inline test/Sentry.Tests/Extensibility/DisabledHubTests.cs:18]: null! ?

Comment (bruno-garcia) [reply test/Sentry.Tests/Extensibility/DisabledHubTests.cs:18]: just to quiet the warning, it doesn't matter here

**Pull request merged**